packages/recommend-core/src/getBatchRecommendations.ts

```diff
@@ -0,0 +1,77 @@
+import {
+  RecommendClient,
+  RecommendationsQuery,
+  TrendingQuery,
+} from '@algolia/recommend';
+
+import { ProductRecord, TrendingFacet } from './types';
+import { mapByScoreToRecommendations, mapToRecommendations } from './utils';
+import { version } from './version';
+
+export type BatchKeyPair = {
+  key: string;
+  value: number;
+};
+
+export type BatchQuery<TObject> = (RecommendationsQuery | TrendingQuery) & {
+  transformItems?: (
+    items: Array<ProductRecord<TObject>>
+  ) => Array<ProductRecord<TObject>>;
+};
+
+export type GetBatchRecommendations<TObject> = {
+  keys: BatchKeyPair[];
+  queries: Array<BatchQuery<TObject>>;
+  recommendClient: RecommendClient;
+};
+
+export type BatchRecommendations<TObject> = {
+  recommendations: Array<ProductRecord<TObject>>;
+};
+
+export async function getBatchRecommendations<TObject>({
+  keys,
+  queries,
+  recommendClient,
+}: GetBatchRecommendations<TObject>): Promise<
+  Record<string, BatchRecommendations<TObject>>
+> {
+  recommendClient.addAlgoliaAgent('recommend-core', version);
+
+  const response = await recommendClient.getRecommendations<TObject>(queries);
+
+  let prevChunks = 0;
+  let allChunks = 0;
+
+  const results: Record<string, BatchRecommendations<TObject>> = {};
+
+  keys.forEach((keyPair) => {
```

Comment:
fine as a forEach, but reduce fits this just as well

Reply:
or a map and Object.fromEntries

Reply:
I know this is a bit subjective, but I found this to be more readable than the more clever reduce approach.